leaf-analysis/src/simulation/CSPLinks.java

```diff
@@ -54,16 +54,17 @@ private static void addForwardContribution(
 				IntVar refTP = CSPPath.getTimePoint(timePointMap, link.getLinkTP());
 	   			for (int t = 0; t < values[targetID].length; t++){
 	   				PrimitiveConstraint preConstraint = null;
-	   				if(pre != null)
+	   				if(pre != null  && pre != ContributionType.NONE)
 	   					preConstraint = createForwardContributionConstraint(pre, values[sourceID][t], values[targetID][t]);
 	   				PrimitiveConstraint postConstraint = null;
-	   				if(post != null)
+	   				if(post != null && post != ContributionType.NONE)
 	   					postConstraint = createForwardContributionConstraint(post, values[sourceID][t], values[targetID][t]);
-	   				if((pre != null) && (post != null))
+	   				
+	   				if(preConstraint != null && postConstraint != null)
 	   					constraints.add(new IfThenElse(new XgtY(refTP, timePoints[t]), preConstraint, postConstraint));
-	   				else if (pre != null)
+	   				else if (preConstraint != null)
```

Comment:
As long as the usage of pre/postConstraint instead of pre and post is intentional think it's good

Reply:
Yup!